### PR TITLE
EZEE-3224: Segmentation API

### PR DESCRIPTION
**Pull request by DominikaK**
| Question      | Answer
| ------------- | ---
| JIRA Ticket   | https://jira.ez.no/browse/EZEE-3224
| Versions      | master (3.2)

Targeting block and managing segments in the Back Office will be covered in a separate PR.

To do:
- [x] Verify examples after all parts of the feature are merged

### CODE REVIEW

File: docs/api/public_php_api_managing_users.md [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    ```
+
+    To create a Segment, use `SegmentationService::createSegment()`
+    and provide it a `SegmentCreateStruct`, which takes the group as one of the parameters:
+
+    ``` php
+    $segmentCreateStruct = new SegmentCreateStruct([
+        'name' => 'Segment 1',
+        'identifier' => 'segment_1',
+        'group' => $newSegmentGroup,
```
Comment (webhdx) [inline docs/api/public_php_api_managing_users.md:78]:
```suggestion
        'group' => $segmentGroup,
```

It should indicate that you need to pass existing group.

**Review by webhdx — APPROVED** [review]
One minor remark

File: docs/api/public_php_api_managing_users.md [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+    ]);
+
+    $newSegment = $this->segmentationService->createSegment($segmentCreateStruct);
+    ```
+
+    ### Updating Segments
+
+    To update a Segment or a Segment Group, use `SegmentationService::updateSegment()`
+    or `SegmentationService::updateSegmentGroup()` and provide it with `SegmentUpdateStruct`
+    or `SegmentGroupUpdateStruct` respectively.
```
Comment (dabrt) [inline docs/api/public_php_api_managing_users.md:88]:
```suggestion
    or `SegmentGroupUpdateStruct`, respectively.
```

File: docs/api/public_php_api_managing_users.md [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+    ### Deleting Segments
+
+    To delete a Segment or a Segment Group, use `SegmentationService::removeSegment()`
+    or `SegmentationService::removeSegmentGroup()` respectively:
```
Comment (dabrt) [inline docs/api/public_php_api_managing_users.md:93]:
```suggestion
    or `SegmentationService::removeSegmentGroup()`, respectively:
```

File: docs/guide/admin_panel.md [hunk 4]
```diff
@@ -263,3 +263,18 @@ By default, eZ Platform contains one Object state group: **Lock**, with states *
 
 Object states can be used in conjunction with permissions, in particular with the [State Limitation](limitation_reference.md#state-limitation).
 Their specific use cases depend on your needs and the setup of your permission system.
+
+## Segments
+
+Segments are used to display specific content to specific Users.
```
Comment (dabrt) [inline docs/guide/admin_panel.md:269]: We might consider going for the active voice here:
"You use segments to display ..."

File: docs/guide/limitation_reference.md [hunk 5]
```diff
@@ -310,6 +310,24 @@ A Limitation to specify if the User has access to content within a specific Sect
 |------|------|------|
 |`<Session_id>`|`<Session_name>`|All valid session IDs can be set as value(s)|
 
+## Segment Group Limitation
+
+A Limitation to specify if the User has access Segments within a specific Segment Group.
```
Comment (dabrt) [inline docs/guide/limitation_reference.md:315]:
```suggestion
A Limitation to specify whether the User has access Segments within a specific Segment Group.
```

File: docs/guide/limitation_reference.md [hunk 6]
```diff
@@ -310,6 +310,24 @@ A Limitation to specify if the User has access to content within a specific Sect
...
+| Value Class     | `Ibexa\Platform\Segmentation\Permission\Limitation\Value\SegmentGroupLimitation` |
+| Type Class      | `Ibexa\Platform\Segmentation\Permission\Limitation\SegmentGroupLimitationType ` |
+| Criterion used  | n/a |
+| Role Limitation | yes                                                                  |
+
+### Possible values
+
+|Value|UI value|Description|
+|------|------|------|
+|`<Segment_group_id>`|`<Segment_group_name>`|All valid Segment Group IDs can be set as value(s)|
```
Comment (dabrt) [inline docs/guide/limitation_reference.md:329]:
```suggestion
|`<Segment_group_id>`|`<Segment_group_name>`|All valid Segment Group IDs can be set as value(s).|
```

**Review by dabrt — CHANGES_REQUESTED** [review]

File: docs/api/public_php_api_managing_users.md [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+    }
+    ```
+
+    Similarly, you can load a Segment in a group using `SegmentationService::loadSegmentGroup()`:
+
+    ``` php
+    $segment = $this->segmentationService->loadSegment(12);
+    ```
+
+    ### Checking assignation
```
[the remark this answers is not in the document]
Comment (DominikaK) [reply docs/api/public_php_api_managing_users.md:32]: Good point

Comment (DominikaK) [inline docs/api/public_php_api_managing_users.md:32]:
```suggestion
    ### Checking assignment
```

File: docs/guide/admin_panel.md (hunk 4 above)
Comment (DominikaK) [inline docs/guide/admin_panel.md:269]:
```suggestion
You can use Segments to display specific content to specific Users.
```

**Review by dabrt — APPROVED** [review]

**Pull request merged**